### ISSUE TITLE
remove JoinStrategy SHUFFLE

### ISSUE THREAD

**Issue by lbergelson** [broadinstitute/gatk#2231]
Shuffle lost the race.  It's slow and buggy (see #2216).  We should remove it.

**Comment by lbergelson** [comment]
@droazen Any objections to this?

**Comment by droazen** [comment]
No, but it should wait until after Tom's code is merged and fully profiled, as it may end up replacing both join strategies.

**Comment by droazen** [comment]
As discussed in https://github.com/broadinstitute/gatk/issues/1203, we are keeping SHUFFLE so that we have a baseline during performance testing.

**Comment by lbergelson** [comment]
@droazen People continue to encounter errors trying to use shuffle, see https://github.com/broadinstitute/dsde-docs/issues/2318.  I repeat my recommendation to remove it.

**Comment by vdauwera** [comment]
Please make this option hidden if it's only being kept for testing purposes (and document clearly that that is the case). Users should not have access to options that are not expected to have value.
